packages/core/components/FileDetails/index.tsx

```diff
@@ -75,20 +75,51 @@ export default function FileDetails(props: Props) {
     const [thumbnailPath, setThumbnailPath] = React.useState<string | undefined>();
     const [isThumbnailLoading, setIsThumbnailLoading] = React.useState(true);
     const stackTokens: IStackTokens = { childrenGap: 12 + " " + 20 };
+    const [calculatedSize, setCalculatedSize] = React.useState<number | null>(null);
+
+    const platformDependentServices = useSelector(
+        interaction.selectors.getPlatformDependentServices
+    );
+    const fileDownloadService = platformDependentServices.fileDownloadService;
+    const isOnWeb = useSelector(interaction.selectors.isOnWeb);
+    const isZarr = fileDetails?.path.endsWith(".zarr") || fileDetails?.path.endsWith(".zarr/");
 
     React.useEffect(() => {
+        setCalculatedSize(null);
         if (fileDetails) {
             setIsThumbnailLoading(true);
             fileDetails.getPathToThumbnail().then((path) => {
                 setThumbnailPath(path);
                 setIsThumbnailLoading(false);
             });
+
+            // Determine size of Zarr on web.
+            if (isOnWeb && isZarr) {
+                if (fileDetails.size && fileDetails.size > 0) {
+                    setCalculatedSize(fileDetails.size);
+                } else {
+                    const { hostname, key } = fileDownloadService.parseS3Url(
+                        fileDetails.downloadPath
+                    );
+                    fileDownloadService
+                        .calculateS3DirectorySize(hostname, key)
+                        .then(setCalculatedSize);
+                }
+            }
         }
-    }, [fileDetails]);
+    }, [fileDetails, fileDownloadService, isOnWeb, isZarr]);
 
     const processStatuses = useSelector(interaction.selectors.getProcessStatuses);
     const openWithMenuItems = useOpenWithMenuItems(fileDetails || undefined);
 
+    // Disable download of large Zarrs ( > 2GB).
+    const isDownloadDisabled = fileDetails
+        ? processStatuses.some((status) => status.data.fileId?.includes(fileDetails.id)) ||
+          (isOnWeb &&
+              isZarr &&
+              (calculatedSize === null || calculatedSize > 2 * 1024 * 1024 * 1024))
```

Comment:
How long does it take to calculate various zarrs? Like how long does it take for a 2GB zarr vs a 1TB zarr?

Reply:
The largest I tested with was a 40Gb file and it was pretty instantaneous.

Reply:
Is this value likely to change/would it make sense to have a MAXIMUM_DOWNLOAD_SIZE const somewhere instead calculating the 2 * 1024 * 1024 * 1024 since we also use it in FileDownloadServiceWeb ?